app/models/manageiq/providers/redhat/infra_manager/refresh_parser.rb

```diff
@@ -496,7 +496,8 @@ def self.vm_inv_to_disk_hashes(inv, storage_uids)
           :present         => true,
           :filename        => device[:id],
           :location        => index.to_s,
-          :size            => device[:size].to_i,
+          :size            => device[:provisioned_size] ? device[:provisioned_size].to_i : device[:size].to_i,
```

Comment:
This can be simplified to `:size            => (device[:provisioned_size] || device[:size]).to_i`